golem/README.md

```diff
@@ -0,0 +1,308 @@
+# ZIO-Golem
+
+[![Scala 3](https://img.shields.io/badge/scala-3.3.x-red.svg)](https://www.scala-lang.org/)
+[![Scala.js](https://img.shields.io/badge/scala.js-1.20.x-blue.svg)](https://www.scala-js.org/)
+[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)
+
+**A minimal, type-safe Scala SDK for building Golem agents.**
+
+ZIO-Golem brings the ergonomics of Scala to the Golem platform, enabling you to define agents as simple traits and
+automatically derive all the serialization, RPC bindings, and metadata generation at compile time via Scala 3 macros.
+
+## Features
+
+- **Trait-based agent definitions** - Define your agent's interface as a Scala trait with annotated methods
+- **Automatic schema derivation** - Derives schemas for component-model serialization
+- **Macro-powered autowiring** - Compile-time generation of RPC handlers, WIT types, and metadata
+- **Multimodal data support** - First-class support for text and binary segments with MIME/language constraints
+- **Transaction helpers** - Both fallible and infallible transaction patterns with automatic rollback
+- **Snapshot integration** - Simple hooks for state persistence across component instances
+
+## Quick Start
+
+### Prerequisites
+
+1. **`golem-cli`** installed and on your `PATH`:
+   ```bash
+   curl -sSf https://join.golem.network/as-requestor | bash -
```

Comment:
What is golem.network?

Reply:
The real one: https://learn.golem.cloud/cli#installation

Reply:
Yeah, that's my bad. I had originally done some AI generation for the initial boilerplate and this never got removed when I cleaned up. I'll get it cut.

Reply:
This should be fixed and I've updated all the docs to properly represent the latest code.